internal/core/data/event_test.go

```diff
@@ -56,7 +55,7 @@ func TestCountByDevice(t *testing.T) {
 	dbClientMock := &dbMock.DBClient{}
 	dbClientMock.On("EventCountByDeviceId", mock.Anything).Return(2, nil)
 
-	count, err := countEventsByDevice(testEvent.Device, context.Background(), dbClientMock)
+	count, err := countEventsByDevice(testEvent.Device, context.Background(), dbClientMock, newMockDeviceClient())
```

Comment:
You're referencing a mock from a different test file albeit from the same package.  This is problematic and couples event_test.go to device_test.go making future refactoring of these tests (and their underlying implementation) to separate packages more difficult.  A more flexible implementation would extract the mock factory function into its own package.

I see you didn't write newMockDeviceClient(); you're simply using it.  

Commented here; applies elsewhere.